mobile/ios/device/ios-core.ts

```diff
@@ -695,15 +695,20 @@ export class WinSocket implements Mobile.IiOSDeviceSocket {
 		});
 	}
 
-	public receiveMessage(): IFuture<Mobile.IiOSSocketResponseData> {
+	public receiveMessage(): IFuture<Mobile.IiOSSocketResponseData | Mobile.IiOSSocketResponseData[]> {
 		return (() => {
-			let message = this.receiveMessageCore();
 			if(this.format === CoreTypes.kCFPropertyListXMLFormat_v1_0) {
-				let reply = plist.parse(message);
+				let message = this.receiveMessageCore();
+				let reply = plist.parse(message.toString());
 				return reply;
 			}
 
-			// TODO: add parsing for binary plists
+			if(this.format === CoreTypes.kCFPropertyListBinaryFormat_v1_0) {
+				// Returns array of results.
+				let message = this.receiveBinaryMessage();
```

Comment:
Not a merge stopper, but the two lines can be shortened to 
```TypeScript
return this.receiveBinaryMessage();
```

without losing clarity IMO